portal-web/docroot/html/taglib/ui/page_iterator/start.jsp

```diff
@@ -326,7 +326,7 @@ NumberFormat numberFormat = NumberFormat.getNumberInstance(locale);
 
 										<li>
 											<liferay-ui:csp>
-												<a class="dropdown-item" href="<%= _getHREF(formName, namespace + curParam, i, jsCall, url, urlAnchor) %>" onclick="<%= forcePost ? _getOnClick(namespace, curParam, i) : "" %>"><span class="sr-only"><liferay-ui:message key="page" /><%= StringPool.NBSP %></span><%= i %></a>
+												<a class="dropdown-item" href="<%= _getHREF(formName, namespace + curParam, i, jsCall, url, urlAnchor) %>" id="<%= String.valueOf(i) %>" onclick="<%= forcePost ? _getOnClick(namespace, curParam, i) : "" %>"><span class="sr-only"><liferay-ui:message key="page" /><%= StringPool.NBSP %></span><%= i %></a>
```

Comment:
Because this (and other tags changed in this file) are enclosed in a `<liferay-ui:csp>` tag, all `id`  values we generate must be unique on the page. Reason is that csp tag rewrites the output to make browser happy with CSP restrictions for inline handlers. Tag uses the `id` value to associate a listener that executes the code inside `onclick` attribute. 

When `id` is not present in the tag, it generates a random one, otherwise reuses it.

As a result, we need to make sure not only these ids are unique within this file but within the whole page. Just using plain numbers might not be as safe as we'd like.

We can solve this with a random number or just prefixing the  `randomNamespace` value calculated on top of the jsp